src/client/components/AddReservation/AddReservation.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from "react-router";
+import { Link } from 'react-router-dom';
+import Moment from 'moment';
+import ReactStars from "react-rating-stars-component";
+import Pics from '../Helper/Pics';
+import OkModal from '../Helper/OkModal'
+
+const AddReservation = (props) => {
+    const params = useParams();
+    const meal = props.meals.filter((m) => m.id == Number(params.id))[0];
+    const avgStars = props.stars.filter((m) => m.id == Number(params.id))[0];
+    const [reservation, setReservation] = useState();
+    const [guests, setGuests] = useState();
+    const date = Moment(new Date()).format('YYYY-MM-DD');
+    const [contact, setContact] = useState("");
+    const [name, setName] = useState("");
+    const [email, setEmail] = useState("");
+    const [availableReservations, setAvailableReservations] = useState(null)
+    const [pics, setPics] = useState(Pics)
+    const [show, setShow] = useState(false)
+
+
+    useEffect(() => {
+        fetch(`api/meals?availableReservations=true`)
+            .then(res => res.json())
+            .then(meals => {
+                const meal1 = meals.filter((m) => m.id === Number(params.id))[0];
+                setAvailableReservations(meal1 ? meal1.No_of_available_reservations : meal.max_reservations)
+            })
+        console.log(reservation);
+        if (reservation) {
```

Comment:
Hmm this reservation parameter seems a bit unnecessary and confusing, wouldn't it be simpler to add a new reservation in the handleSubmit function?

Reply:
Yes .. i will try to make it simpler